src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,37 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
+// ______________________________//
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
+// ______________________________//
+
 class Carnivore extends Animal {
-  // write your code here
+  bite(bittenAnimal) {
+    if (bittenAnimal.hidden !== true && bittenAnimal instanceof Herbivore) {
```

Comment:
Ensure that `bittenAnimal` is an instance of `Animal` before proceeding. Consider adding a check like `bittenAnimal instanceof Animal` to prevent errors if a non-animal object is passed.